File: rest-api/src/main/java/tg/scam/detector/entity/BanInfo.java
package tg.scam.detector.entity;

import com.fasterxml.jackson.databind.JsonNode;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import org.springframework.data.rest.core.annotation.RestResource;

import java.util.UUID;

@Data
@Entity
@NoArgsConstructor
@Table(name = "ban_list")
public class BanInfo {
    @Id
    @Column(name = "ban_id")
    private UUID id = UUID.randomUUID();
    @OneToOne
    @JoinColumn(name = "user_id")
    @RestResource(exported = false)
    private User user;
    private String cause;
    @JdbcTypeCode(SqlTypes.JSON)
    private JsonNode proofs;
    @ManyToOne
    @JoinColumn(name="owner_id")
    @RestResource(exported = false)
    private User owner;

    public BanInfo(User user, String cause) {
        this.user = user;
        this.cause = cause;
    }
}

File: tg-bot/core/api/entities/ban_info.py
class BanInfo:
    def __init__(self, uuid: str, user_id: str, owner_id: str, cause: str, proofs: dict):
        self.uuid = uuid
        self.user_id = user_id
        self.owner_id = owner_id
        self.cause = cause
        self.proofs = proofs

    def to_params(self) -> dict:
        from core.api.request import Requests
        return {
            Requests.BanParams.ID: self.uuid,
            Requests.BanParams.USER: {
                Requests.BanParams.USER_ID: self.user_id
            },
            Requests.BanParams.OWNER_ID: {
                Requests.BanParams.USER_ID: self.owner_id
            },
            Requests.BanParams.CAUSE: self.cause,
            Requests.BanParams.PROOFS: self.proofs
        }
